engine/src/flutter/common/input/text_input_connection.h

```diff
@@ -0,0 +1,44 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_COMMON_INPUT_TEXT_INPUT_CONNECTION_H_
+#define FLUTTER_COMMON_INPUT_TEXT_INPUT_CONNECTION_H_
+
+#include <string>
+#include "fml/closure.h"
+
+namespace flutter {
+
+class TextInputConnection {
+ public:
+  TextInputConnection() = default;
+
+  virtual ~TextInputConnection() {}
+
+  virtual std::string GetCurrentText() = 0;
+
+  virtual void SetCurrentText(std::string_view text) = 0;
```

Comment:
Do you know if we would suffer any performance overhead by calling this from framework side to read and write to the state in embedding? probably fine for iOS, but I am more worry about android since everything need to go through jni. thus serialization